RSSReader/View/SearchViewController.m

```diff
@@ -0,0 +1,158 @@
+        _searchBar.translatesAutoresizingMaskIntoConstraints = NO;
+        _searchBar.placeholder = @"Enter site name";
+        _searchBar.layer.borderWidth = 1;
+        _searchBar.delegate = self;
+        _searchBar.layer.borderColor = [UIColor colorWithRed:240.0/255.0 green:240.0/255.0 blue:241.0/255.0 alpha:1].CGColor;
+    }
+    return _searchBar;
+}
+
+- (UITableView *)tableView {
+    if (!_tableView) {
+        _tableView = [[UITableView alloc] initWithFrame:CGRectZero style:UITableViewStyleGrouped];
+        _tableView.translatesAutoresizingMaskIntoConstraints = NO;
+        _tableView.dataSource = self;
+        _tableView.delegate = self;
+        [_tableView registerClass:UITableViewCell.class forCellReuseIdentifier:@"cellId"];
+    }
+    return _tableView;
+}
+
+#pragma mark UITableViewDataSource implementation
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.itemsArray.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cellId"];
+    cell.textLabel.text = self.itemsArray[indexPath.row].title;
+    return cell;
+}
+
+- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
+    return @"Results";
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+    [tableView deselectRowAtIndexPath:indexPath animated:YES];
+    [self saveChoiseAtIndexPath:indexPath];
+    [self createAndPushFeedViewController:indexPath];
+}
+
+- (void)createAndPushFeedViewController:(NSIndexPath *)indexPath {
+    NSMutableArray *feedItemArray = [NSMutableArray new];
+    FeedViewController *feed = [[FeedViewController alloc] initWithTitle:self.itemsArray[indexPath.row].title];
+    feed.feedItemArray = feedItemArray;
+    RSSParser *parser = [RSSParser new];
+    FeedPresenter *presenter = [[FeedPresenter alloc] initWithArray:feedItemArray networkManager:[NetworkManager sharedInstance] parser:parser url:self.itemsArray[indexPath.row].url];
+    feed.presenter = presenter;
+    [parser release];
+    [presenter release];
+    [self.navigationController pushViewController:feed animated:false];
+    [feed release];
+    [feedItemArray release];
+}
+
+#pragma mark UISearchBarDelegate implementation
+
+- (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar {
+    [self showActivityIndicator];
+    NSMutableArray *itemsArray = [NSMutableArray new];
+    self.itemsArray = itemsArray;
+    __block typeof(self) weakSelf = self;
+    [self.presenter searchFeeds:searchBar.text array:itemsArray completion:^{
+        [weakSelf hideActivityIndicator];
+        [weakSelf.tableView reloadData];
+    }];
+    [itemsArray release];
+}
+
+- (void)showException:(NSString *)message {
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Error"
```

Comment:
Please use NSLocalizedString for strings which are displayed to user.

Reply:
done